eureka-client/src/main/java/com/netflix/discovery/shared/Applications.java

```diff
@@ -252,201 +237,91 @@ public String getReconcileHashCode() {
     }
 
     /**
-     * Populates the provided instance count map.  The instance count map is used as part of the general
-     * app list synchronization mechanism.
-     * @param instanceCountMap the map to populate
+     * Populates the provided instance count map. The instance count map is used
+     * as part of the general app list synchronization mechanism.
+     * 
+     * @param instanceCountMap
+     *            the map to populate
      */
-    public void populateInstanceCountMap(TreeMap<String, AtomicInteger> instanceCountMap) {
+    public void populateInstanceCountMap(Map<String, AtomicInteger> instanceCountMap) {
         for (Application app : this.getRegisteredApplications()) {
             for (InstanceInfo info : app.getInstancesAsIsFromEureka()) {
-                AtomicInteger instanceCount = instanceCountMap.get(info.getStatus().name());
-                if (instanceCount == null) {
-                    instanceCount = new AtomicInteger(0);
-                    instanceCountMap.put(info.getStatus().name(), instanceCount);
-                }
+                AtomicInteger instanceCount = instanceCountMap.computeIfAbsent(info.getStatus().name(),
+                        k -> new AtomicInteger(0));
                 instanceCount.incrementAndGet();
             }
         }
     }
 
     /**
-     * Gets the reconciliation hashcode.  The hashcode is used to determine whether the applications list
-     * has changed since the last time it was acquired.
-     * @param instanceCountMap the instance count map to use for generating the hash
+     * Gets the reconciliation hashcode. The hashcode is used to determine
+     * whether the applications list has changed since the last time it was
+     * acquired.
+     * 
+     * @param instanceCountMap
+     *            the instance count map to use for generating the hash
      * @return the hash code for this instance
      */
-    public static String getReconcileHashCode(TreeMap<String, AtomicInteger> instanceCountMap) {
-        String reconcileHashCode = "";
+    public static String getReconcileHashCode(Map<String, AtomicInteger> instanceCountMap) {
+        StringBuilder reconcileHashCode = new StringBuilder(75);
```

Comment:
some maths :). 
If you sum the lengths of each member in InstanceStatus (35), add in 2 * InstanceStatus.length for the separators that the hashing implementation adds (10), and figure an max size of 6 characters for the stringified 'count' associated with each status (25), then you get 76.  75 was subjectively used instead because fives.